### PR TITLE
faust: replace qt4 with qt5

### PR DESCRIPTION
**Pull request by magnetophon**
###### Description of changes

Replaces https://github.com/NixOS/nixpkgs/pull/180387
See also https://github.com/NixOS/nixpkgs/pull/174634:  "qt4: remove"

To test:

Create a simple faust program:
```bash
echo "process = _,_; " >> tst.dsp
```
Compile and run it.  This needs access to an alsa sound card, so you need to have either pipewire running or you need bare alsa, iow you need to stop pulse for a moment.
```bash
faust2alqt tst.dsp && ./tst
```
This one needs either pipewire or jack to be running:
```bash
faust2jaqt tst.dsp && ./tst
```

The third script; ``faust2lv2``, works too, but not fully.
It produces a plugin, to be loaded inside another program (usually a daw).
This plugin can be compiled either without a GUI, which is the default and works fine, or with a QT GUI. 
The plugin with a GUI works as well, but only if you "manually wrap" the calling program, like so:
```bash
QT_PLUGIN_PATH=${QT_PLUGIN_PATH:+':'$QT_PLUGIN_PATH':'}

QT_PLUGIN_PATH=${QT_PLUGIN_PATH/':''/nix/store/imd975l7lpqx4y9479yn6bp7vgkgzmzn-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins'':'/':'}
QT_PLUGIN_PATH='/nix/store/imd975l7lpqx4y9479yn6bp7vgkgzmzn-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins'$QT_PLUGIN_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH#':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH%':'}
export QT_PLUGIN_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH:+':'$QT_PLUGIN_PATH':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH/':''/nix/store/0y0y7sqakvbi4k15ji4q17czbq652jl6-qtsvg-5.15.8-bin/lib/qt-5.15.8/plugins'':'/':'}
QT_PLUGIN_PATH='/nix/store/0y0y7sqakvbi4k15ji4q17czbq652jl6-qtsvg-5.15.8-bin/lib/qt-5.15.8/plugins'$QT_PLUGIN_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH#':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH%':'}
export QT_PLUGIN_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH:+':'$QT_PLUGIN_PATH':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH/':''/nix/store/jldpa244glc8y66lz77kyz6xk349bciv-qtdeclarative-5.15.8-bin/lib/qt-5.15.8/plugins'':'/':'}
QT_PLUGIN_PATH='/nix/store/jldpa244glc8y66lz77kyz6xk349bciv-qtdeclarative-5.15.8-bin/lib/qt-5.15.8/plugins'$QT_PLUGIN_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH#':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH%':'}
export QT_PLUGIN_PATH
QML2_IMPORT_PATH=${QML2_IMPORT_PATH:+':'$QML2_IMPORT_PATH':'}
QML2_IMPORT_PATH=${QML2_IMPORT_PATH/':''/nix/store/jldpa244glc8y66lz77kyz6xk349bciv-qtdeclarative-5.15.8-bin/lib/qt-5.15.8/qml'':'/':'}
QML2_IMPORT_PATH='/nix/store/jldpa244glc8y66lz77kyz6xk349bciv-qtdeclarative-5.15.8-bin/lib/qt-5.15.8/qml'$QML2_IMPORT_PATH
QML2_IMPORT_PATH=${QML2_IMPORT_PATH#':'}
QML2_IMPORT_PATH=${QML2_IMPORT_PATH%':'}
export QML2_IMPORT_PATH
QML2_IMPORT_PATH=${QML2_IMPORT_PATH:+':'$QML2_IMPORT_PATH':'}
QML2_IMPORT_PATH=${QML2_IMPORT_PATH/':''/nix/store/6vk5p8x3kars7v9siimmqmcsiw0vvlbc-qtquickcontrols-5.15.8/lib/qt-5.15.8/qml'':'/':'}
QML2_IMPORT_PATH='/nix/store/6vk5p8x3kars7v9siimmqmcsiw0vvlbc-qtquickcontrols-5.15.8/lib/qt-5.15.8/qml'$QML2_IMPORT_PATH
QML2_IMPORT_PATH=${QML2_IMPORT_PATH#':'}
QML2_IMPORT_PATH=${QML2_IMPORT_PATH%':'}
export QML2_IMPORT_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH:+':'$QT_PLUGIN_PATH':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH/':''/nix/store/ba6wycyx4kh690iiwifppdfi2qbii2ks-qtwayland-5.15.8-bin/lib/qt-5.15.8/plugins'':'/':'}
QT_PLUGIN_PATH='/nix/store/ba6wycyx4kh690iiwifppdfi2qbii2ks-qtwayland-5.15.8-bin/lib/qt-5.15.8/plugins'$QT_PLUGIN_PATH
QT_PLUGIN_PATH=${QT_PLUGIN_PATH#':'}
QT_PLUGIN_PATH=${QT_PLUGIN_PATH%':'}
export QT_PLUGIN_PATH
QML2_IMPORT_PATH=${QML2_IMPORT_PATH:+':'$QML2_IMPORT_PATH':'}
QML2_IMPORT_PATH=${QML2_IMPORT_PATH/':''/nix/store/ba6wycyx4kh690iiwifppdfi2qbii2ks-qtwayland-5.15.8-bin/lib/qt-5.15.8/qml'':'/':'}
QML2_IMPORT_PATH='/nix/store/ba6wycyx4kh690iiwifppdfi2qbii2ks-qtwayland-5.15.8-bin/lib/qt-5.15.8/qml'$QML2_IMPORT_PATH
QML2_IMPORT_PATH=${QML2_IMPORT_PATH#':'}
QML2_IMPORT_PATH=${QML2_IMPORT_PATH%':'}
export QML2_IMPORT_PATH
exec -a "$0" $the_calling_program
```

Any ideas on how to make this work without the manual wrapping?
@ajs124 @pmahoney Wanna take a look? Thanks!

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GabrielDougherty** [comment]
Result of `nixpkgs-review pr 215561` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mooSpace</li>
    <li>tambura</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2csound</li>
    <li>faust2jack</li>
    <li>faust2jackrust</li>
    <li>faust2jaqt</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
  </ul>
</details>

Also faust2alqt worked with manual testing.

File: pkgs/applications/audio/faust/faust2.nix
Comment (K900) [inline pkgs/applications/audio/faust/faust2.nix:204]: Not necessary anymore.

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (K900) [inline pkgs/applications/audio/faust/faust2alqt.nix:28]: Isn't this just `wrapQtAppsHook`?

Comment (ajs124) [reply pkgs/applications/audio/faust/faust2alqt.nix:28]: I think we had some trouble getting that to work in #180387, but it probably is, yes.

Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2alqt.nix:28]: ``wrapQtAppsHook`` would wrap the faust compiler, we want to wrap the output of the compiler.

File: pkgs/applications/audio/faust/faust2.nix
Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2.nix]: Works without as well, indeed.

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (K900) [reply pkgs/applications/audio/faust/faust2alqt.nix:31]: Is the compiler something that runs at _runtime_? So this package is just the compiler, and it's its outputs that need wrapping?

Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2alqt.nix:31]: Yes.

Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2alqt.nix:32]: I fixed the directory issue.

**Comment by magnetophon** [comment]
@K900 I think this is good to merge now.

**Comment by GabrielDougherty** [comment]
Result of `nixpkgs-review pr 215561` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages failed to build:</summary>
  <ul>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mooSpace</li>
    <li>tambura</li>
    <li>zrythm</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>faust2alqt</li>
    <li>faust2jaqt</li>
    <li>faust2lv2</li>
  </ul>
</details>

**Comment by magnetophon** [comment]
@GabrielDougherty Thanks, I got most of them working again.
I marked ``ConstantDetuneChorus`` as broken, since I'm still figuring out what went wrong there.
I don't understand how ``zrythm`` is related, since there is no faust dependency in it.
When I build it with ``nix-env -f $NIXPKGS -iA zrythm``, it works just fine, but it errors out when built as part of ``nixpkgs-review pr --print-result 215561``.

**Comment by magnetophon** [comment]
What is holding this back from being merged?

**Comment by geri1701** [comment]
Result of `nixpkgs-review pr 215561` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>zrythm</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>faust2alqt</li>
    <li>faust2jaqt</li>
    <li>faust2lv2</li>
    <li>faustPhysicalModeling</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mooSpace</li>
    <li>tambura</li>
  </ul>
</details>

**Comment by magnetophon** [comment]
faustStk does build with the latest faust, https://github.com/NixOS/nixpkgs/pull/231700

Result of `nixpkgs-review pr 231700` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>faustStk</li>
  </ul>
</details>
<details>
  <summary>29 packages built:</summary>
  <ul>
    <li>faust</li>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2csound</li>
    <li>faust2firefox</li>
    <li>faust2jack</li>
    <li>faust2jackrust</li>
    <li>faust2jaqt</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>faust2sc</li>
    <li>faustPhysicalModeling</li>
    <li>faustlive</li>
    <li>guitarix</li>
    <li>kapitonov-plugins-pack</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mooSpace</li>
    <li>open-music-kontrollers.mephisto</li>
    <li>tambura</li>
    <li>zrythm</li>
  </ul>
</details>

**Review by GabrielDougherty — APPROVED** [review]

**Comment by GabrielDougherty** [comment]
LGTM

> What is holding this back from being merged?

Getting the attention of someone with merge access

**Review by geri1701 — APPROVED** [review]

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:37875]: This is missing an entry in aliases.nix

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2jaqt.nix:31]:
```suggestion
  wrapBinary = ''
```

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2jaqt.nix:51]:
```suggestion
  preFixup = ''
    for script in "$out"/bin/*; do
      # append the wrapping code to the compilation script
      echo "${wrapBinary}" >> $script
      # prevent the qmake error when running the script
      sed -i "/QMAKE=/c\ QMAKE="${qtbase.dev}/bin/qmake"" $script
    done
```

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2jaqt.nix:11]: we could put wrapBinary into a let in rather than using rec here

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2alqt.nix:41]:
```suggestion
  preFixup = ''
    for script in "$out"/bin/*; do
      # append the wrapping code to the compilation script
      echo "${wrapBinary}" >> $script
      # prevent the qmake error when running the script
      sed -i "/QMAKE=/c\ QMAKE="${qtbase.dev}/bin/qmake"" $script
    done
```

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2alqt.nix:20]:
```suggestion
  wrapBinary = ''
```

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2alqt.nix:23]:
```suggestion
      workpath=$PWD
```

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2alqt.nix:27]:
```suggestion
      makeWrapper "./.$binary-wrapped" ".$binary-temp" --set QT_PLUGIN_PATH "${qtbase}/${qtbase.qtPluginPrefix}"
```

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (SuperSandro2000) [inline pkgs/applications/audio/faust/faust2alqt.nix:28]:
```suggestion
      mv ".$binary-temp" "$binary"
```

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2jaqt.nix:51]: Thanks for the review!
When I try these suggestions regarding wrapping, I end up with:
```bash
source /nix/store/7w5cmms8sh5pv70xgz2ysmhgdi12nwyf-make-shell-wrapper-hook/nix-support/setup-hook
for p in ; do
  workpath=/build/source
  cd -- .
  binary=
  mv  .-wrapped
  makeWrapper ./.-wrapped .-temp  --set QT_PLUGIN_PATH /nix/store/rb7gbnldym6dphxam2znp9kiw3flzw8n-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins
  mv .-temp
  sed -i  -e 's@exec@cd /nix/store/c3f4jdwzn8fm9lp72m91ffw524bakp6v-stdenv-linux \&\& exec@g'
  cd
done
```
While I need:
```bash
source /nix/store/7w5cmms8sh5pv70xgz2ysmhgdi12nwyf-make-shell-wrapper-hook/nix-support/setup-hook
for p in $FILES; do
  workpath=$(pwd)
  cd -- "$(dirname "$p")"
  binary=$(basename --suffix=.dsp "$p")
  mv $binary ."$binary"-wrapped
  makeWrapper ./."$binary"-wrapped ."$binary"-temp  --set QT_PLUGIN_PATH "/nix/store/rb7gbnldym6dphxam2znp9kiw3flzw8n-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins"
  mv ."$binary"-temp $binary
  sed -i $binary -e 's@exec@cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" \&\& exec@g'
  cd $workpath
done
```

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2alqt.nix:15]: I implemented your suggestions as far as I was able to.
Thanks again for the review @SuperSandro2000!

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2jaqt.nix]: That leaves me with:
```bash
source /nix/store/l84drzijhk1ilnn3yvv2r42wh2lj43ni-make-shell-wrapper-hook/nix-support/setup-hook
for p in $FILES; do
  workpath=$PWD
  cd -- "$(dirname "$p")"
  binary=$(basename --suffix=.dsp "$p")
  mv $binary ."$binary"-wrapped
  makeWrapper "./.$binary-wrapped" ".$binary-temp" --set QT_PLUGIN_PATH "/nix/store/rzp9gihz7rfg9l59gjpzavy2wrg7vjdj-qtbase-5.15.9-bin/lib/qt-5.15.9/plugins"
  mv ".$binary-temp" "$binary"
  sed -i $binary -e s@exec@cd /nix/store/smcpmkfc7bx90ry7wmr6nckg83anjj0k-stdenv-linux && exec@g
  cd $workpath
done
```

File: pkgs/applications/audio/faust/faust2alqt.nix
Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2alqt.nix]: I cleaned up this part too now.
When you use wrapProgram, the double ``mv`` is not needed anymore.

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (SuperSandro2000) [reply pkgs/applications/audio/faust/faust2jaqt.nix]: So only the bash source part is wrong. If you can't solve that on your own, we can also keep it like it is, to make it less brittle on changes. You decide.

File: pkgs/top-level/aliases.nix
Comment (Mindavi) [inline pkgs/top-level/aliases.nix:449]: Nit: missing space

File: pkgs/applications/audio/magnetophonDSP/ConstantDetuneChorus/default.nix
Comment (Mindavi) [inline pkgs/applications/audio/magnetophonDSP/ConstantDetuneChorus/default.nix:37]: Remove, maybe? Or will this be fixed in the future?

Comment (magnetophon) [reply pkgs/applications/audio/magnetophonDSP/ConstantDetuneChorus/default.nix:37]: This is working again in the new faust version: https://github.com/NixOS/nixpkgs/pull/231700.

File: pkgs/applications/audio/faust/faust2jaqt.nix
Comment (magnetophon) [reply pkgs/applications/audio/faust/faust2jaqt.nix]: I think the source part is OK, it's mainly the end that is wrong.
I need:
```bash
source /nix/store/7w5cmms8sh5pv70xgz2ysmhgdi12nwyf-make-shell-wrapper-hook/nix-support/setup-hook
for p in $FILES; do
  workpath=$(pwd)
  cd -- "$(dirname "$p")"
  binary=$(basename --suffix=.dsp "$p")
  mv $binary ."$binary"-wrapped
  makeWrapper ./."$binary"-wrapped ."$binary"-temp  --set QT_PLUGIN_PATH "/nix/store/rb7gbnldym6dphxam2znp9kiw3flzw8n-qtbase-5.15.8-bin/lib/qt-5.15.8/plugins"
  mv ."$binary"-temp $binary
  sed -i $binary -e 's@exec@cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" \&\& exec@g'
  cd $workpath
done
```

**Comment by PowerUser64** [comment]
Tested with the following procedure:
```bash
nix-shell -p nixpkgs-review --run "nixpkgs-review pr 215561"
echo "process = _,_; " >> tst.dsp
faust2jaqt tst.dsp && ./tst
```
This succeeds, with the only package reported to fail building being `zrythm`, which was updated just yesterday (#231486).

**Pull request merged**
